scipy/optimize/_highs/meson.build

```diff
@@ -11,7 +11,8 @@ highs_define_macros = [
   '-UEXT_PRESOLVE',
   '-USCIP_DEV',
   '-UHiGHSDEV',
-  '-UOSI_FOUND'
+  '-UOSI_FOUND',
+  '-DNDEBUG'
```

Comment:
Above it says `'-DCMAKE_BUILD_TYPE="RELEASE"'`, which confuses me relative to the matching issue description: 

> I received a tip that HiGHS may be compiled in our 1.9.3 release with debug asserts, e.g. -DCMAKE_BUILD_TYPE=RelWithDebug

In fact, even if the issue is correct and `RelWithDebug` was used, both normally already have `-DNDEBUG` unless it is being overridden somewhere: https://stackoverflow.com/a/59314670/2942522

Reply:
`-DCMAKE_BUILD_TYPE=RelWithDebug` and `'-DCMAKE_BUILD_TYPE="RELEASE"'` don't actually do anything here because we're not using cmake.  My recollection is that we added those originally at the beginning to try to define all the same things.  AFAIK, they are not used anywhere